app/src/main/java/ru/kryu/playlistmaker/search/data/TrackHistoryRepositoryImpl.kt

```diff
@@ -1,24 +1,43 @@
 package ru.kryu.playlistmaker.search.data
 
-import ru.kryu.playlistmaker.search.data.storage.mapper.TrackForStorageToDomain
-import ru.kryu.playlistmaker.search.data.storage.mapper.TrackToTrackForStorage
+import ru.kryu.playlistmaker.search.data.storage.mapper.MapperTrackForStorage
 import ru.kryu.playlistmaker.search.domain.api.TrackHistoryRepository
 import ru.kryu.playlistmaker.search.domain.model.Track
 
 class TrackHistoryRepositoryImpl(private val historyStorage: HistoryStorage) :
     TrackHistoryRepository {
-    override fun getTrackHistory(): List<Track> {
+
+    private val trackHistory = getTrackHistoryFromStorage()
```

Comment:
🍏 Чтобы все время не кастить к MutableList, можно сразу привести значение к MutableList.
Еще нормальная практика - заводить две переменные, одна из который приватная и мутабельная, а вторая публичная и немутабельная. Первая используется только внутри, а вторую можно отдавать наружу.
Например:
```suggestion
    private val _trackHistory = getTrackHistoryFromStorage().toMutableList()
    val trackHistory = _trackHistory.toList()
```